### PR TITLE
Support experimental enabling of multiple bucket spaces.

### PR DESCRIPTION
**Pull request by geirst**
@vekterli please review
@toregge FYI

### CODE REVIEW

File: config-model/src/main/java/com/yahoo/vespa/model/content/GlobalDistributionValidator.java [hunk 1]
```diff
@@ -22,9 +22,11 @@
...
                          Set<NewDocumentType> globallyDistributedDocuments,
-                         Redundancy redundancy) {
-
-        verifyGlobalDocumentsHaveRequiredRedundancy(globallyDistributedDocuments, redundancy);
+                         Redundancy redundancy,
+                         boolean enableMultipleBucketSpaces) {
+        if (!enableMultipleBucketSpaces) {
+            verifyGlobalDocumentsHaveRequiredRedundancy(globallyDistributedDocuments, redundancy);
+        }
         verifySearchableCopiesIsSameAsRedundancy(globallyDistributedDocuments, redundancy);
```
Comment (vekterli) [inline config-model/src/main/java/com/yahoo/vespa/model/content/GlobalDistributionValidator.java]: Shouldn't this check also be ignored when multiple bucket spaces are enabled? The global distribution config transform will ensure that number of ready copies == redundancy == number of nodes, but this property will not necessarily hold for the node group(s) specified in the XML.

**Review by vekterli — COMMENTED** [review]
Let's discuss searchable copies validation first

**Comment by geirst** [comment]
@vekterli please review again

**Review by vekterli — APPROVED** [review]
:+1:

**Pull request merged**